src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/UriQueryGenerator.cs

```diff
@@ -17,7 +18,50 @@ internal class UriQueryGenerator : SearchParamTableExpressionQueryGenerator
 
         public override SearchParameterQueryGeneratorContext VisitString(StringExpression expression, SearchParameterQueryGeneratorContext context)
         {
-            return VisitSimpleString(expression, context, VLatest.UriSearchParam.Uri, expression.Value);
+            switch (expression.FieldName)
+            {
+                case FieldName.Uri:
+                    return VisitSimpleString(expression, context, VLatest.UriSearchParam.Uri, expression.Value);
+                case FieldName.UriVersion:
+                    return VisitSimpleString(expression, context, VLatest.UriSearchParam.Version, expression.Value);
+                case FieldName.UriFragment:
+                    return VisitSimpleString(expression, context, VLatest.UriSearchParam.Fragment, expression.Value);
+                default:
+                    throw new NotSupportedException();
+            }
+        }
+
+        public override SearchParameterQueryGeneratorContext VisitMissingField(MissingFieldExpression expression, SearchParameterQueryGeneratorContext context)
+        {
+            switch (expression.FieldName)
+            {
+                case FieldName.UriVersion:
+                    return VisitMissingFieldImpl(expression, context, FieldName.UriVersion, VLatest.UriSearchParam.Fragment);
+                case FieldName.UriFragment:
+                    return VisitMissingFieldImpl(expression, context, FieldName.UriFragment, VLatest.UriSearchParam.Fragment);
+                default:
+                    throw new NotSupportedException();
+            }
+        }
+
+        public override SearchParameterQueryGeneratorContext VisitBinary(BinaryExpression expression, SearchParameterQueryGeneratorContext context)
+        {
+            if (expression.BinaryOperator != BinaryOperator.Equal)
+            {
+                throw new NotSupportedException();
+            }
+
+            switch (expression.FieldName)
+            {
+                case FieldName.Uri:
+                    return VisitSimpleBinary(BinaryOperator.Equal, context, VLatest.UriSearchParam.Uri, null, expression.Value);
```

Comment:
preserve the componentIndex

Reply:
I've removed VisitBinary as this wasn't being used anymore